app/src/main/java/com/csc/telezhnaya/todo2/TaskManager.java

```diff
@@ -0,0 +1,125 @@
+                Date date = new Date(cursor.getLong(cursor.getColumnIndexOrThrow(COLUMN_DATE)));
+                taskDate.setText(date.toString());
+                boolean done = cursor.getInt(cursor.getColumnIndex(COLUMN_STATUS)) == 1;
+                checkBox.setChecked(done);
+                star.setChecked(cursor.getInt(cursor.getColumnIndex(COLUMN_STARRED)) == 1);
+                int id = cursor.getInt(cursor.getColumnIndexOrThrow(TaskTable._ID));
+                view.setTag(id);
+
+                if (done) {
+                    taskName.setPaintFlags(taskName.getPaintFlags() | Paint.STRIKE_THRU_TEXT_FLAG);
+                    taskDate.setPaintFlags(taskName.getPaintFlags() | Paint.STRIKE_THRU_TEXT_FLAG);
+                } else {
+                    taskName.setPaintFlags(taskName.getPaintFlags() & ~Paint.STRIKE_THRU_TEXT_FLAG);
+                    taskDate.setPaintFlags(taskName.getPaintFlags() & ~Paint.STRIKE_THRU_TEXT_FLAG);
+                }
+            }
+        };
+    }
+
+    public void unbind() {
+        context = null;
+    }
+
+    public CursorAdapter getAdapter() {
+        return adapter;
+    }
+
+    public String getDescription(int position) {
+        Cursor cursor = context.getContentResolver().query(ContentUris.withAppendedId(TASKS_URI, position),
+                null, null, null, null);
+        cursor.moveToNext();
+        String description = cursor.getString(cursor.getColumnIndex(TaskTable.COLUMN_DESCRIPTION));
+        cursor.close();
+        return description;
+    }
+
+    public void addTask(String text) {
+        long time = new Date().getTime();
+        ContentValues values = new ContentValues();
+        values.put(COLUMN_TEXT, text);
+        values.put(COLUMN_DATE, time);
+        values.put(COLUMN_STATUS, 0);
+        values.put(COLUMN_STARRED, 0);
+        context.getContentResolver().insert(TASKS_URI, values);
+    }
+
+    public void updateTask(int position, String newText, String newDescription, boolean done, boolean starred) {
+        ContentValues values = new ContentValues();
+        values.put(COLUMN_TEXT, newText);
+        if (newDescription != null) {
+            values.put(COLUMN_DESCRIPTION, newDescription);
+        }
+        values.put(COLUMN_STATUS, done ? 1 : 0);
+        values.put(COLUMN_STARRED, starred ? 1 : 0);
+        context.getContentResolver().update(ContentUris.withAppendedId(TASKS_URI, position),
+                values, TaskTable._ID + "=?", new String[]{String.valueOf(position)});
+    }
+
+    @Override
+    public Loader<Cursor> onCreateLoader(int id, Bundle args) {
+        return new CursorLoader(context, TASKS_URI, null, null, null, TaskManager.TASK_ORDER);
+    }
+
+    @Override
+    public void onLoadFinished(Loader<Cursor> loader, Cursor data) {
+        // насколько я смогла разобраться, нужно вызывать changeCursor, чтобы старый убивали
+        adapter.changeCursor(data);
```

Comment:
I have read somewhere that swapCursor does not close old cursor.
I am not sure in that source.
Could you please tell the difference between changeCursor and swapCursor? It would be very helpful for me.